PMMain/Midas/MidasGUI/RingData.h

```diff
@@ -0,0 +1,34 @@
+#include <vector>
+#pragma once
+class ringData
+{
+public:
+    ringData();
+    ~ringData();
+
+    void Dispose_object()
+    {
+        delete this;
+    }
+
+    struct keyboard_value
+    {
+        char main, hold;
+        keyboard_value(char mainVal = '\0', char holdVal = '\0') : main(mainVal), hold(holdVal) { }
+    };
```

Comment:
mainVal shouldnt have a default value. Rather, it should be mandatory to populate it.